extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -460,17 +660,102 @@ export class SchemaCompareDialog {
-			|| (!this.targetIsDacpac && !isNullOrUndefined(this.targetDatabaseDropdown.value) && this.targetDatabaseDropdown.values.findIndex(x => this.matchesValue(x, this.targetDbEditable)) !== -1);
+			|| (this.targetIsProject && this.existsProjectFile(this.targetTextBox.value))
+			|| (this.targetIsDatabase && !isNullOrUndefined(this.targetDatabaseDropdown.value) && this.targetDatabaseDropdown.values.findIndex(x => this.matchesValue(x, this.targetDbEditable)) !== -1);
 
 		return sourcefilled && targetfilled;
 	}
 
+	public async handleOkButtonClick(): Promise<void> {
+		await this.execute();
+		this.dispose();
+	}
+
+	protected showErrorMessage(message: string): void {
+		this.dialog.message = {
+			text: message,
+			level: getAzdataApi()!.window.MessageLevel.Error
+		};
+	}
+
+	async validate(): Promise<boolean> {
+		try {
+
+			if ((this.sourceIsProject || this.targetIsProject) && !vscode.extensions.getExtension(loc.sqlDatabaseProjectExtensionId)) {
+				this.showErrorMessage(loc.noProjectExtension);
+				return false;
+			}
+
+			if (this.sourceIsProject && this.targetIsProject) {
+				try {
+					await this.getDsp(true);
+				} catch (eSource) {
+					try {
+						await this.getDsp(false);
+					} catch (eTarget) {
+						this.showErrorMessage(loc.dspErrorSourceAndTarget);
+						return false;
+					}
+
+					this.showErrorMessage(loc.dspErrorSource);
+					return false;
+				}
+			}
+
+			if (this.sourceIsProject) {
+				try {
+					await this.getDsp(true);
+				} catch (eSource) {
+					this.showErrorMessage(loc.dspErrorSource);
+					return false;
+				}
+			}
+
+			if (this.targetIsProject) {
+				try {
+					await this.getDsp(false);
+				} catch (eSource) {
+					this.showErrorMessage(loc.dspErrorTarget);
+					return false;
+				}
+			}
+
+			return true;
+		} catch (e) {
+			this.showErrorMessage(e?.message ? e.message : e);
+			return false;
+		}
+	}
+
+	private dispose(): void {
+		this.toDispose.forEach(disposable => disposable.dispose());
+	}
+
 	private async existsDacpac(filename: string): Promise<boolean> {
 		return !isNullOrUndefined(filename) && await exists(filename) && (filename.toLocaleLowerCase().endsWith('.dacpac'));
 	}
 
+	private async existsProjectFile(filename: string): Promise<boolean> {
+		return !isNullOrUndefined(filename) && await exists(filename) && (filename.toLocaleLowerCase().endsWith('.sqlproj'));
+	}
+
+	private async getTargetScripts(source: boolean): Promise<string[]> {
+		const projectFilePath = source ? this.sourceTextBox.value : this.targetTextBox.value;
+		return await vscode.commands.executeCommand(loc.sqlDatabaseProjectsGetTargetScripts, projectFilePath);
+	}
+
+	private async getDsp(source: boolean): Promise<string> {
+		try {
+			const projectFilePath = source ? this.sourceTextBox.value : this.targetTextBox.value;
+			return await vscode.commands.executeCommand(loc.sqlDatabaseProjectsGetDsp, projectFilePath);
+		} catch (e) {
+			throw e;
```

Comment:
why have a try catch if the error is just going to be thrown again and nothing else happens?